AppAuth.podspec

```diff
@@ -33,16 +33,26 @@ It follows the OAuth 2.0 for Native Apps best current practice
   s.platforms    = { :ios => "7.0", :osx => "10.9", :watchos => "2.0", :tvos => "9.0" }
 
   s.source       = { :git => "https://github.com/openid/AppAuth-iOS.git", :tag => s.version }
-
-  s.source_files = "Source/*.{h,m}"
   s.requires_arc = true
 
-  # iOS
-  s.ios.source_files      = "Source/iOS/**/*.{h,m}"
-  s.ios.deployment_target = "7.0"
-  s.ios.frameworks        = "SafariServices", "AuthenticationServices"
-
-  # macOS
-  s.osx.source_files = "Source/macOS/**/*.{h,m}"
-  s.osx.deployment_target = '10.9'
+  # Subspec for the core AppAuth library classes only, suitable for extensions.
+  s.subspec 'Core' do |core|
+     core.source_files = "Source/*.{h,m}"
+     core.exclude_files = "Source/AppAuth.h"
```

Comment:
When installing `pod 'AppAuth'` without specifying any subspec, I understood that it takes all subspecs if there is no `default_subspec` defined.

But the tests shows that `AppAuth.h` is included, so I guess this line gets overridden by the ExternalUserAgent subspec. Seems good 🙂.

Reply:
Yes, this was my theory as well. The two statements are taken together to build the list of files, so the exclusion only applies at the sub-spec level.

We could actually make ExternalUserAgent the default since it's completely stand-alone. The official docs recommend that the main spec include all the sub-specs which is why I didn't. I guess it doesn't really make much difference.

If you look at the xcodeworkspace with the sample, the ExternalUserAgent folder just contains the files that AppAuthCore doesn't, which seems to be the right outcome there at least.